File: 0242-valid-anagram/0242-valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t):
            return False
        a = ''.join(sorted(s))
        b = ''.join(sorted(t))
        print(a)
        print("hello :", b)
        if a == b:
            return True
        return False

File: 0242-valid-anagram/0242-valid-anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
       // Brute Force Approach
        // sort(s.begin() , s.end());
        // sort(t.begin() , t.end());
        // if(s==t){
        //     return true;
        // }
        // else{
        //     return false;
        // }


        // Optimized 
        if (s.length() != t.length()) {
            return false;
        }
        
        vector<int> freq(26, 0);
        for (int i = 0; i < s.length(); i++) {
            freq[s[i] - 'a']++;
            freq[t[i] - 'a']--;
        }
        
        for (int i = 0; i < freq.size(); i++) {
            if (freq[i] != 0) {
                return false;
            }
        }
        
        return true;
    }
};